server/game/cards/12-KotI/BalonGreyjoy.js

```diff
@@ -0,0 +1,29 @@
+const DrawCard = require('../../drawcard.js');
+
+class BalonGreyjoy extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Put card into play',
+            cost: ability.costs.kneelFactionCard(),
+            target: {
+                cardCondition: card => card.location === 'discard pile' && card.controller !== this.controller &&
+                                       this.controller.canPutIntoPlay(card)
```

Comment:
As discussed in chat, this may not actually be the case, but this should be fine until we get a clear ruling one way or the other.